bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -875,6 +881,20 @@ public void run() {
     @Override
     public void modifiedService(ServiceReference/* <ModuleHandlerFactory> */ reference,
             /* ModuleHandlerFactory */Object service) {
+        logger.debug("ModuleHandlerFactory modified, updating handlers");
+        ModuleHandlerFactory moduleHandlerFactory = ((ModuleHandlerFactory) service);
+        @SuppressWarnings("unchecked")
+        Collection<String> removed_handlers = (Collection<String>) reference.getProperty("removed_handlers");
```

Comment:
> This could for example result in the following:
> 
> 1.    Someone has made a complex action in a scripting language
> 2.   After some time, he thinks this Action would be nice to share.  He can than modularize it easily, ....

Hi Simon,
the above scenario fits very well with the Composite module types. Please check their definition in the doc file:
https://github.com/eclipse/smarthome/blob/master/docs/documentation/features/rules.md#composite-module-types
So the user can create a new composite module type extending the ScriptAction with predefined "script" config parameter